src/game/Strategic/Strategic_Mines.cc

```diff
@@ -701,29 +660,32 @@ void SaveMineStatusToSaveGameFile(HWFILE const f)
 
 void LoadMineStatusFromSavedGameFile(HWFILE const f)
 {
-	FOR_EACH(MINE_STATUS_TYPE, i, gMineStatus)
+	//FOR_EACH(MINE_STATUS_TYPE, i, gMineStatus)
+	// TODO: what if number of mines changed/
```

Comment:
Which way is your worry? That the save has more or less than what's in the externalized data?

Reply:
Both ways break the save.  The number of mines in save must match exactly the JSON data.

Reply:
I guess warn and abort if it happens, then?

Reply:
There is no annotation or metadata in save game data. It is just a bunch of bytes.

There is no way to be sure if we are reading valid data of a min, or "over-reading" into the next section of data.

Reply:
That's too bad then, I thought the various sections would be accessible by offset fields that we could then check against. You can still add a warning for non-default values though.

Reply:
Warning message added in MineModel::validateData